conan/tools/cmake/cmake.py

```diff
@@ -209,3 +213,19 @@ def test(self, build_type=None, target=None, cli_args=None, build_tool_args=None
         env = ["conanbuild", "conanrun"] if env == "" else env
         self._build(build_type=build_type, target=target, cli_args=cli_args,
                     build_tool_args=build_tool_args, env=env)
+
+    @property
+    def _verbosity(self):
+        verbosity = self._conanfile.conf.get("tools.build:verbosity")
+        if verbosity:
+            # ERROR, WARNING, NOTICE, STATUS (default), VERBOSE, DEBUG, or TRACE
+            verbosities = {
+
+            }
+            if verbosity not in ("quiet", "error", "warning", "notice", "status", "normal",
```

Comment:
lets add a ``conf.get("myconf", choices=iterable-of-choices)`` to do the validation inside ``conf.get``, in the same way we check the return type? Because it seems this pattern is common enough

Reply:
👍 good idea

Reply:
I think that implementation should have a dedicated PR and keep this PR only fixing what it's supposed to fix.

From my point of view, I think that `choices` should have some kind of predefined fixed values to make them automatically checked (apart from this development which would be necessary as well). For instance, `verbosity`, `compiler_executables`, `microsoft.bash:subsystem`, etc.  could have those fixed values, won't they? Then, it could be better to have something like:


```python
BUILT_IN_CONFS = {
"conf_1": ("doc", DEFAULT, [CHOICE_1, CHOICE_2]),
...
}
```

I'm not saying that this implementation should be the only one but it could be a proposal.

Reply:
I like your idea about having the type be part of the conf definition, I have some questions about how it would interact with non-built-in confs, but this is something we can discuss :)

Reply:
Non-built-in confs would use your proposal I guess (`conf.get("xxxx", choices=())`), but sure, we can discuss it later.